components/intl/DateFormatter.tsx

```diff
@@ -0,0 +1,40 @@
+/*
+ * Sonar UI Common
+ * Copyright (C) 2019-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+import * as React from 'react';
+import { DateSource, FormattedDate } from 'react-intl';
+import { parseDate } from '../../helpers/dates';
+
+export interface DateFormatterProps {
+  children?: (formattedDate: string) => React.ReactNode;
+  date: DateSource;
+  long?: boolean;
+}
+
+export const formatterOption = { year: 'numeric', month: 'short', day: '2-digit' };
+
+export const longFormatterOption = { year: 'numeric', month: 'long', day: 'numeric' };
```

Comment:
How about SHOUTCASING (uppercasing) these? Or does that imply too much change in `sonar-web`?

Reply:
I think we shouldn't export these, and have a formatDate function that takes a `long` prop.. but I'm hesitant to do it

Reply:
In what cases are these used? I know I used it for the leak period components. Don't remember the details, though.

Reply:
mostly when including a formatted date in a `translateWithParams` I think.
But I think we should use a `FormattedMessage` with the appropriate component rather than using these options manually

Reply:
I looked quickly and the first one I found is in `CreationDateFacet`, and it wouldn't be easy (possible?) to switch for a component there...
The formatDate function seems like a good idea but I wonder how you'd get it to work outside react since it must use the one that is inside the intl context ?